junit5-engine/src/main/java/org/junit/gen5/engine/junit5/execution/injection/MethodArgumentResolver.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2015 the original author or authors.
+ *
+ * All rights reserved. This program and the accompanying materials are
+ * made available under the terms of the Eclipse Public License v1.0 which
+ * accompanies this distribution and is available at
+ *
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.junit.gen5.engine.junit5.execution.injection;
+
+import java.lang.reflect.*;
+
+import org.junit.gen5.commons.util.*;
+
+// for a 'real' solution see: org.springframework.web.method.support.HandlerMethodArgumentResolver
+public interface MethodArgumentResolver {
+
+	boolean supports(Parameter parameter);
```

Comment:
I think it's fine to use `Parameter` for now but I would rather pass in our own abstraction of it so we can make sure annotations are read consistently etc. in a next step.